sklearn/model_selection/tests/test_validation.py

```diff
@@ -915,54 +916,157 @@ def test_cross_val_predict_sparse_prediction():
+                                         method=method, cv=kfold)
+    assert_array_equal_maybe_list(predictions, predictions_ystr)
 
-        # Test alternative representations of y
-        predictions_y1 = cross_val_predict(est, X, y + 1, method=method,
-                                           cv=kfold)
-        assert_array_equal(predictions, predictions_y1)
+    return predictions
 
-        predictions_y2 = cross_val_predict(est, X, y - 2, method=method,
-                                           cv=kfold)
-        assert_array_equal(predictions, predictions_y2)
 
-        predictions_ystr = cross_val_predict(est, X, y.astype('str'),
-                                             method=method, cv=kfold)
-        assert_array_equal(predictions, predictions_ystr)
+def assert_array_equal_maybe_list(x, y):
+    # If x and y are lists of arrays, compare arrays individually.
+    if isinstance(x, list):
+        for i in range(len(x)):
+            assert_array_equal(x[i], y[i])
+    else:
+        assert_array_equal(x, y)
 
 
 def test_cross_val_predict_with_method():
-    check_cross_val_predict_with_method(LogisticRegression())
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
+    for method in ['decision_function', 'predict_proba', 'predict_log_proba']:
+        est = LogisticRegression()
+        out = check_cross_val_predict_with_method(est, X, y, method)
+        assert_array_equal(out.shape, (len(X), len(set(y))))
 
 
 def test_gridsearchcv_cross_val_predict_with_method():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
     est = GridSearchCV(LogisticRegression(random_state=42),
                        {'C': [0.1, 1]},
                        cv=2)
-    check_cross_val_predict_with_method(est)
+    for method in ['decision_function', 'predict_proba', 'predict_log_proba']:
+        out = check_cross_val_predict_with_method(est, X, y, method)
+        assert_array_equal(out.shape, (len(X), len(set(y))))
+
+
+def test_cross_val_predict_with_method_multilabel_ovr():
+    # OVR does multilabel predictions, but only arrays of
+    # binary indicator columns. The output of predict_proba
+    # is a 2D array with shape (n_samples, n_classes).
+    n_samp = 100
+    n_classes = 4
+    X, y = make_multilabel_classification(n_samples=n_samp, n_labels=3,
+                                          n_classes=n_classes, n_features=5,
+                                          random_state=42)
+    est = OneVsRestClassifier(LogisticRegression(random_state=0))
+    for method in ['predict_proba', 'decision_function']:
+        out = check_cross_val_predict_with_method(est, X, y, method=method)
+        assert_array_equal(out.shape, (n_samp, n_classes))
+
+
+def test_cross_val_predict_with_method_multilabel_rf():
+    # The RandomForest allows anything for the contents of the labels.
```

Comment:
The wording here is unclear. Do you just mean that RF handles multiclass-multioutput and produces `predict_proba` in that vein?